proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,601 @@
+# Machine readable JSON output for dotnet list package
+
+* Status: **In Review**
+* Author: [Erick Yondon](https://github.com/erdembayar)
+* GitHub Issue [7752](https://github.com/NuGet/Home/issues/7752)
+
+## Problem background
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many organizations are required by [regulation](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) to audit packages that they're using in a repository.
+
+Currently there's no easy way to produce a [Software Bill of Material (SBOM)](https://blog.sonatype.com/what-is-a-software-bill-of-materials) output which can be consumed by another auditing system or kept for records.
+
+* Parse-friendly output. Other package managers like [NPM already have it](https://docs.npmjs.com/cli/v7/commands/npm-ls) (`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * Check any vulnerable packages
+    * Check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issues (detect duplicate packages, newly introduced dependencies, dependency removals, etc.)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### `--format` option
+
+Ability to use new `--format` option for all `dotnet list package` commands to ensure formatted(JSON etc) output is emitted to the console.
+
+```dotnetcli
+dotnet list [<PROJECT>|<SOLUTION>] package [--config <SOURCE>]
+    [--deprecated]
+    [--framework <FRAMEWORK>] [--highest-minor] [--highest-patch]
+    [--include-prerelease] [--include-transitive] [--interactive]
+    [--outdated] [--source <SOURCE>] [-v|--verbosity <LEVEL>]
+    [--vulnerable]
+    [--format <FORMAT>]
+
+dotnet list package -h|--help
+```
+
+`<FORMAT>` - Allowed values as part of spec is `json`. (In the future `parseable`, `csv`, `yaml`, `xml` could be candidates.)
```

Comment:
Should the default format (i.e. text) be one of the options here?

Reply:
`plain text` can be passed to.

Reply:
I just included.